authorization/resource/resource.go

```diff
@@ -0,0 +1,223 @@
+package resource
+
+import (
+	"context"
+	"time"
+
+	"github.com/fabric8-services/fabric8-auth/account"
+	"github.com/fabric8-services/fabric8-auth/gormsupport"
+	"github.com/jinzhu/gorm"
+	"github.com/fabric8-services/fabric8-auth/application/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/log"
+
+	"github.com/goadesign/goa"
+	errs "github.com/pkg/errors"
+	"github.com/satori/go.uuid"
+)
+
+type Resource struct {
+	gormsupport.Lifecycle
+
+	// This is the primary key value
+	ID string `sql:"type:string" gorm:"primary_key" gorm:"column:resource_id"`
+	// The parent resource
+	ParentResource *Resource
+	// The owning identity
+	Owner account.Identity
+	// The resource type
+	ResourceType ResourceType
+	// Resource description
+	Description string
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m Resource) TableName() string {
+	return "resource"
+}
+
+// GetLastModified returns the last modification time
+func (m Resource) GetLastModified() time.Time {
+	return m.UpdatedAt
+}
+
+// GormResourceRepository is the implementation of the storage interface for Resource.
+type GormResourceRepository struct {
+	db *gorm.DB
+}
+
+// NewResourceRepository creates a new storage type.
+func NewResourceRepository(db *gorm.DB) *GormResourceRepository {
+	return &GormResourceRepository{db: db}
+}
+
+// ResourceRepository represents the storage interface.
+type ResourceRepository interface {
+	repository.Exister
+	Load(ctx context.Context, id string) (*Resource, error)
+	Create(ctx context.Context, resource *Resource) error
+	Lookup(ctx context.Context, id string) (*Resource, error)
+	Save(ctx context.Context, resource *Resource) error
+	Delete(ctx context.Context, id string) error
+	Query(funcs ...func(*gorm.DB) *gorm.DB) ([]Resource, error)
+	List(ctx context.Context) ([]Resource, error)
```

Comment:
There is no List() implementation which makes sense. So, remove List() from interface too.